### PR TITLE
Implement the fourth wave of per-reconciler leaderelection

### PR DESCRIPTION
**Pull request by mattmoor**
Detailed design: https://docs.google.com/document/d/1i_QHjQO2T3SNv49xjZLWlivcc0UvZN1Tbw2NKxThkyM/edit

Issue: #1181

WIP until #1303 lands

### CODE REVIEW

**Comment by mattmoor** [comment]
/retest

🤔

File: codegen/cmd/injection-gen/generators/reconciler_reconciler.go [hunk 1]
```diff
@@ -303,27 +381,34 @@ func (r *reconcilerImpl) Reconcile(ctx {{.contextContext|raw}}, key string) erro
...
-		{{end}}
+			// Reconcile this copy of the resource and then write back any status
+			// updates regardless of whether the reconciliation errored out.
+			reconcileEvent = r.reconciler.ReconcileKind(ctx, resource)
 
-		// Reconcile this copy of the resource and then write back any status
-		// updates regardless of whether the reconciliation errored out.
-		reconcileEvent = r.reconciler.ReconcileKind(ctx, resource)
+			{{if .isKRShaped}}
+			reconciler.PostProcessReconcile(ctx, resource, original)
```
Comment (whaught) [inline codegen/cmd/injection-gen/generators/reconciler_reconciler.go:402]: looks like the krshaped pre/post process are properly preserved! Thanks for  the merge resolution

Comment (whaught) [reply codegen/cmd/injection-gen/generators/reconciler_reconciler.go:402]: /lgtm

Comment (mattmoor) [reply codegen/cmd/injection-gen/generators/reconciler_reconciler.go:402]: cool thanks

**Comment by dprotaso** [comment]
FYI - there's a hold - unsure about the context

**Comment by mattmoor** [comment]
Context for `/hold` is that I wanted @n3wscott to TAL, since he's been driving most of the codegen work (and was out Friday).

**Pull request merged**